frontend/src/components/bucket/BucketConfigDialog.vue

```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
+import Dialog from 'primevue/dialog';
+import BucketConfigForm from '@/components/bucket/BucketConfigForm.vue';
+import type { Bucket } from '@/interfaces';
+
+const props = defineProps<{
+  display: boolean;
+  header: string;
+  title: string;
+  isUpdate: boolean;
+  bucket?: Bucket;
+}>();
+
+const emit = defineEmits(['close-bucket-config']);
+
+const closeDialog = () => {
+  emit('close-bucket-config');
+};
+</script>
+
+<template>
+  <div>
+    <Dialog
+      :visible="props.display"
+      :style="{width: '50vw'}"
+      :modal="true"
+      @update:visible="closeDialog"
+    >
+      <template #header>
+        <div class="flex">
+          <font-awesome-icon
+            icon="fas fa-cog"
+            class="pr-3 pt-2"
+            style="font-size: 2rem"
+          />
+          <div>
+            <h1>{{ header }}</h1>
+            <h3>{{ title }}</h3>
+          </div>
+        </div>
+      </template>
+      <BucketConfigForm
+        :is-update="props.isUpdate"
+        :bucket="props.bucket"
+        @submit-bucket-config="closeDialog"
+        @cancel-bucket-config="closeDialog"
+      />
+    </Dialog>
+  </div>
```

Comment:
Unnecessary div wrapper - as it only wraps one DOM element, we can remove it and instead have the Dialog element be the top level template element instead.
Also generally wondering whether we need to be breaking down this dialog into its own component or not as this isn't a very reusable component and very specific to the bucket context.

Reply:
div removed.

Reply:
I would refactor this to have this entire <Dialog> directly in the BucketList instead of an extra level of redirection, and delete the BucketConfigDialog.vue file.

Reply:
Personally, I am not a big fan of big files/components (100+ lines) - would rather brake down to smaller pieces - easier for testing too. If @jujaga and @TimCsaky agree to refactor in that way, sure no problem. Please bear in mind this will take additional time to test again and etc.

Reply:
not too bothered. i like it as is.

Reply:
I'm going to agree with @kamorel on this. While it's fine making more smaller components, they should really only be done if these components have some degree of reuse outside of the one parent component calling it. Since this is very focused to Bucket addition flow only, there's no merit splitting this apart as you then have to do a bunch of extra watchers and emit listeners to be able to keep state synced between parent and child.

Reply:
Roger that. Refactoring.